### DISCUSSION TITLE
get entity count from engine through code

### DISCUSSION THREAD

**Discussion by ghost** [stride3d/stride#1345] [Q&A] [answered]
in this section of the manual: https://doc.stride3d.net/4.0/en/manual/engine/entity-component-model/managing-entities.html
it mentions a way to loop through all current entities, however the EntityManager doesn't have a descriptor called "Entities", so i'm not sure what to do. i'm new to C#, so i'm struggling to even do somewhat simple things at the moment, sorry if it's a stupid question.

**Comment by VaclavElias** [comment, marked as answer]
Hello @JustAnotherGameDevWasTaken,

C# is not difficult to learn but obviously everything takes some time to learn 😀. There are nice tutorials out there.. e.g. https://dotnet.microsoft.com/en-us/learn/csharp

Regarding your question, I think there are 2 ways.

```c#
 public class YourScript : StartupScript
    {
        public override void Start()
        {
            var sceneEntities1 = Entity.Scene.Entities;
            var sceneEntities2 = SceneSystem.SceneInstance.RootScene.Entities;
        }
    }
```

**Reply by ghost** [reply]
thanks for the quick response, the link to the learning resources is also appreciated!

**Reply by VaclavElias** [reply]
You are welcome. I am actually planning to do similar C# learning resources for beginners + Stride, so one can learn C# with some Stride / Game Development basics at the same time. I will ping you once there is something available.

**Reply by ghost** [reply]
that's actually what i'm doing at the moment, i want to develop a game using a unity like program, but i don't like the whole 15% royalties or whatever that unity charges, so i'm using stride because it seems like a great game engine, plus, c# is a transferrable skill. i'm learning c# specifically for stride game development. so any resources you could provide would be amazingly helpful.
